src/transformers/models/rag/retrieval_rag.py

```diff
@@ -363,13 +363,14 @@ class RagRetriever:
 
     """
 
-    def __init__(self, config, question_encoder_tokenizer, generator_tokenizer, index=None, init_retrieval=True):
+    def __init__(self, config, question_encoder_tokenizer, ctx_encoder_tokenizer,generator_tokenizer, index=None, init_retrieval=True):
```

Comment:
This is a big breaking change -> the ctx_encoder_tokenizer was not required previously so it should only be added optionally

Reply:
oh my bad. We use a set function.

Reply:
@patrickvonplaten  added a set_function that gets called from the distributed retriever class.